owncloudApp/src/main/java/com/owncloud/android/presentation/spaces/SpacesListFragment.kt

```diff
@@ -118,9 +129,25 @@ class SpacesListFragment : SpacesListAdapter.SpacesListAdapterListener, Fragment
 
     override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
         super.onCreateOptionsMenu(menu, inflater)
+        (menu.findItem(R.id.action_search).actionView as SearchView).run {
+            maxWidth = Int.MAX_VALUE
+            queryHint = resources.getString(R.string.actionbar_search)
+            setOnQueryTextListener(this@SpacesListFragment)
+        }
         menu.findItem(R.id.action_share_current_folder)?.itemId?.let { menu.removeItem(it) }
     }
 
+    override fun onQueryTextSubmit(query: String?): Boolean = false
+
+    override fun onQueryTextChange(newText: String?): Boolean {
+        newText?.let { spacesListViewModel.updateSearchFilter(it) }
+        return true
+    }
+
+    fun setSearchListener(searchView: SearchView) {
+        searchView.setOnQueryTextListener(this)
+    }
```

Comment:
If we finally conclude that the call from `FileDisplayActivity` is not needed, this method is not necessary anymore 👍

Reply:
Check previous comments ⬆️